### ISSUE TITLE
xdg-open <folder> opens folder in VSCode instead of file browser since Ubuntu 24

### ISSUE THREAD

**Issue by AlCalzone** [microsoft/vscode#229020]
Type: <b>Bug</b>

I recently upgraded from Ubuntu 22 to 24. Since then, trying to open a folder in the file explorer from VSCode's integrated terminal using `xdg-open /path/to/folder` results in another VSCode window opening for the folder.

If I type the same in the system terminal, it works as expected.

VS Code version: Code 1.93.1 (38c31bc77e0dd6ae88a4e9cc93428cc27a56ba40, 2024-09-11T17:20:05.685Z)
OS version: Linux x64 6.8.0-44-generic snap
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|11th Gen Intel(R) Core(TM) i7-1185G7 @ 3.00GHz (8 x 4298)|
|GPU Status|2d_canvas: unavailable_software<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: disabled_software<br>multiple_raster_threads: enabled_on<br>opengl: disabled_off<br>rasterization: disabled_software<br>raw_draw: disabled_off_ok<br>skia_graphite: disabled_off<br>video_decode: disabled_software<br>video_encode: disabled_software<br>vulkan: disabled_off<br>webgl: unavailable_software<br>webgl2: unavailable_software<br>webgpu: disabled_off<br>webnn: disabled_off|
|Load (avg)|1, 1, 1|
|Memory (System)|31.06GB (21.14GB free)|
|Process Argv|--no-sandbox --force-user-env --crash-reporter-id 23d56de3-001d-431f-81e8-0c352e7ea82f|
|Screen Reader|no|
|VM|14%|
|DESKTOP_SESSION|ubuntu|
|XDG_CURRENT_DESKTOP|Unity|
|XDG_SESSION_DESKTOP|ubuntu|
|XDG_SESSION_TYPE|x11|
</details><details><summary>Extensions (92)</summary>

Extension|Author (truncated)|Version
---|---|---
toggle-excluded-files|amo|2.0.0
tsl-problem-matcher|amo|0.6.2
vscode-zipfs|arc|3.0.0
chronicler|arc|0.1.16
search-crates-io|bel|1.2.1
lit-html|bie|1.11.1
vscode-tailwindcss|bra|0.12.10
vscode-diff-viewer|cap|1.5.0
ruff|cha|2024.48.0
css-theme-completions|con|0.0.5
esbuild-problem-matchers|con|0.0.3
vscode-eslint|dba|3.0.10
dprint|dpr|0.16.3
gitlens|eam|15.5.1
EditorConfig|Edi|0.16.4
RunOnSave|eme|0.2.0
json-tools|eri|1.0.2
prettier-vscode|esb|11.0.0
codespaces|Git|1.17.3
copilot|Git|1.229.0
copilot-chat|Git|0.20.1
vscode-github-actions|git|0.26.5
vscode-pull-request-github|Git|0.88.1
vscode-graphql-syntax|Gra|1.3.6
vscode-mocha-test-adapter|hbe|2.14.1
vscode-test-explorer|hbe|2.21.1
rest-client|hum|0.25.1
cortex-debug|mar|1.12.1
debug-tracker-vscode|mcu|0.0.15
memory-view|mcu|0.0.25
peripheral-viewer|mcu|1.4.6
rtos-views|mcu|0.0.7
template-string-converter|meg|0.6.1
git-graph|mhu|1.30.0
compare-folders|mos|0.24.3
vscode-json5|mrm|1.0.0
vscode-azureresourcegroups|ms-|0.9.5
vscode-docker|ms-|1.29.2
vscode-language-pack-de|MS-|1.93.2024091109
vscode-dotnet-runtime|ms-|2.1.5
debugpy|ms-|2024.10.0
pylint|ms-|2023.10.1
python|ms-|2024.14.1
vscode-pylance|ms-|2024.9.1
jupyter|ms-|2024.8.1
jupyter-keymap|ms-|1.1.2
jupyter-renderers|ms-|1.0.19
vscode-jupyter-cell-tags|ms-|0.1.9
vscode-jupyter-slideshow|ms-|0.1.6
remote-containers|ms-|0.384.0
remote-ssh|ms-|0.114.3
remote-ssh-edit|ms-|0.86.0
remote-wsl|ms-|0.88.3
vscode-remote-extensionpack|ms-|0.25.0
azure-account|ms-|0.12.0
cmake-tools|ms-|1.19.51
cpptools|ms-|1.21.6
cpptools-extension-pack|ms-|1.3.0
hexeditor|ms-|1.10.0
live-server|ms-|0.4.15
makefile-tools|ms-|0.11.13
powershell|ms-|2024.2.2
remote-explorer|ms-|0.4.3
remote-server|ms-|1.5.2
test-adapter-converter|ms-|0.1.9
vscode-js-profile-flame|ms-|1.0.9
vscode-serial-monitor|ms-|0.13.1
hide-gitignored|npx|1.1.0
vetur|oct|0.37.3
vscode-twoslash-queries|Ort|1.2.2
vscode-versionlens|pfl|1.14.2
typescript-mono-repo-import-helper|q|0.0.6
tsserver-live-reload|rbu|1.0.1
java|red|1.34.0
vscode-sort-json|ric|1.20.0
lit-plugin|run|1.4.3
rust-analyzer|rus|0.3.2011
gitconfig|sid|2.0.1
even-better-toml|tam|0.19.2
cmake|twx|0.0.17
sort-lines|Tyr|1.11.0
intellicode-api-usage-examples|Vis|0.2.8
vscodeintellicode|Vis|1.3.1
explorer|vit|1.2.8
vscode-arduino|vsc|0.7.1
vscode-gradle|vsc|3.16.4
vscode-java-dependency|vsc|0.24.0
vscode-java-pack|vsc|0.29.0
vscode-maven|vsc|0.44.0
vscode-icons|vsc|12.9.0
vscode-todo-highlight|way|1.0.5
config-editor|zwa|0.0.15

(1 theme extensions excluded)

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vspor879:30202332
vspor708:30202333
vspor363:30204092
vswsl492:30256859
vscod805cf:30301675
binariesv615:30325510
vsaa593:30376534
py29gd2263:31024239
c4g48928:30535728
azure-dev_surveyone:30548225
a9j8j154:30646983
962ge761:30959799
pythongtdpath:30769146
pythonnoceb:30805159
asynctok:30898717
pythonmypyd1:30879173
h48ei257:31000450
pythontbext0:30879054
dsvsc016:30899300
dsvsc017:30899301
dsvsc018:30899302
cppperfnew:31000557
dsvsc020:30976470
pythonait:31006305
dsvsc021:30996838
9c06g630:31013171
a69g1124:31058053
dvdeprecation:31068756
dwnewjupyter:31046869
impr_priority:31102340
nativerepl1:31139838
refactort:31108082
pythonrstrctxt:31112756
flighttreat:31134774
wkspc-onlycs-t:31132770
wkspc-ranged-t:31125599
defaultse:31133495
ei213698:31121563

```

</details>

**Comment by deepak1556** [comment]
Does the issue repro with our debian package ?

**Comment by AlCalzone** [comment]
Yes and no. For some reason that one opens the folder in VLC media player 🤷🏻‍♂️

**Comment by petervansark** [comment]
Same issue:

Version: 1.94.2
OS: Linux x64 6.8.0-45-generic snap
